frontend/src/components/elements/Video/Video.tsx

```diff
@@ -39,6 +39,11 @@ export default function Video({ element, width }: VideoProps): ReactElement {
     const setStartTime: () => void = () => {
       if (videoNode) {
         videoNode.currentTime = element.startTime
+        /* height of HTML video element needs to be set to avoid scrolling issue: https://github.com/streamlit/streamlit/issues/5069
+           initially HTML video has height 0 and when tabs are switched page auto scrolls up
+           because of "lack of content" on the page, setting height of HTML video element fixes it
+         */
+        videoNode.height = videoNode.videoHeight
```

Comment:
For youtube videos, it seems that we already set the height based on `width * 0.75`. And in my test app, I have used youtube URLs. So I would assume that it should already work at least for Youtube videos. Do you know why it doesn't?:
```
import streamlit as st

for i in range(20):
    st.markdown("dummy text")

tab1, tab2 = st.tabs(["tab1", "tab2"])

with tab1:
    st.video("https://www.youtube.com/watch?v=R2nr1uZ8ffc")

with tab2:
    st.video("https://www.youtube.com/watch?v=R2nr1uZ8ffc")
```

Reply:
btw. I think it would be better here if we move this line outside of the `setStartTime` method. For example:

```
useEffect(() => {
   const videoNode = videoRef.current
   if (videoNode) {
      videoNode.height = videoNode.videoHeight
  }
```

Reply:
> For youtube videos, it seems that we already set the height based on width * 0.75
The problem is not on YouTube videos, but on videos referenced by file handler, for example this code will reproduce the issue:
```
import streamlit as st

for i in range(20):
    st.markdown("dummy text")


tab1, tab2 = st.tabs(["tab1", "tab2"])

video_file = open('earth.mp4', 'rb')
video_bytes = video_file.read()

with tab1:
    st.video(video_bytes)

with tab2:
    st.video(video_bytes)
```

> I think it would be better here if we move this line outside of the setStartTime method. 
@LukasMasuch but the thing is we want to setup the height after `loadedmetadata` event, I can rename `setStartTime` to `onLoadedMetadata`

Reply:
> the thing is we want to setup the height after loadedmetadata event,

Indeed 👍  In that case renaming `setStartTime` is a good suggestion!